nighres/brain/dots_segmentation.py

```diff
@@ -0,0 +1,719 @@
+                   np.nansum(np.exp(g0*U),axis=3))
+    return posterior_l
+    
+
+def dots_segmentation(tensor_image, mask, atlas_dir, wm_atlas = 1, 
+                      max_iter = 25, convergence_threshold = 0.005, s_I = 1/42, 
+                      c_O = 0.5, max_angle = 67.5, save_data = False, 
+                      overwrite = False, output_dir = None, file_name = None):
+    """DOTS segmentation
+
+    Segment major white matter tracts in diffusion tensor images using Diffusion
+    Oriented Tract Segmentation (DOTS) algorithm.
+    
+    Parameters
+    ----------
+    tensor_image: niimg
+        Input image containing the diffusion tensor coefficients in the
+        following order: volumes 0-5: D11, D22, D33, D12, D13, D23
+    mask: niimg
+        Binary brain mask image which limits computation to the defined volume.
+   atlas_dir: str, optional
+        Path to atlas directory where there is a subdirectory called 
+        'DOTS_atlas' which was generated by nighres.data.download_DOTS_atlas()
+    wm_atlas: int, optional
+        Define which white matter atlas to use. Option 1 for 23 tracts [2]_ 
+        and option 2 for 39 tracts [1]_. (default is 1)
+    max_iter: int, optional
+        Maximum number of iterations in the conditional modes algorithm.
+        (default is 20)
+    convergence_threshold: float, optional
+        Threshold for when the iterated conditonal modes algorithm is considered
+        to have converged. Defined as the fraction of labels that change during
+        one step of the algorithm. (default is 0.002)
+    s_I: float, optional
+        Parameter controlling how isotropic label energies propagate to their
+        neighborhood. (default is 1/42)
+    c_O: float, optional
+        Weight parameter for unclassified white matter atlas prior. (default
+        is 1/2)
+    max_angle: float, optional
+        Maximum angle (in degrees) between principal tensor directions before 
+        connectivity coefficient c becomes negative. Possible values between 0
+        and 90. (default is 67.5)
+    save_data: bool, optional
+        Save output data to file. (default is False)
+    overwrite: bool, optional
+        Overwrite existing results. (default is False)
+    output_dir: str, optional
+        Path to desired output directory, will be created if it doesn't exist.
+    file_name: str, optional
+        Desired base name for output files without file extension, suffixes 
+        will be added.
+
+    Returns
+    ----------
+    dict
+        Dictionary collecting outputs under the following keys
+        (type of output files in brackets)
+
+        * segmentation (array_like): Hard segmentation of white matter.
+        * posterior (array_like): POsterior probabilities of tracts.
+        * energy (array_like): Markov Random Field energies of labels.
+        
+    Notes
+    ----------
+    Algorithm details can be found in [1]_.
```

Comment:
Unfortunately you cannot use the footnote reference twice (it screws up the rendering in the built docs). Easy way out for now is to write "Algorithm details can be found in the references below"